addons/medical_treatment/functions/fnc_treatmentBandageLocal.sqf

```diff
@@ -14,92 +14,33 @@
 
 private _openWounds = _target getVariable [QEGVAR(medical,openWounds), []];
 if (_openWounds isEqualTo []) exitWith { false };
 
-// Get the default effectiveness for the used bandage
-private _config = configFile >> QUOTE(ADDON) >> "Bandaging";
-private _effectiveness = getNumber (_config >> "effectiveness");
-
-if (isClass (_config >> _bandage)) then {
-    _config = (_config >> _bandage);
-
-    if (isNumber (_config >> "effectiveness")) then {
-        _effectiveness = getNumber (_config >> "effectiveness");
-    };
-};
-
 // Figure out which injury for this bodypart is the best choice to bandage
 // TODO also use up the remainder on left over injuries
-private _mostEffectiveSpot = 0;
-private _effectivenessFound = -1;
-private _mostEffectiveInjury = _openWounds select 0;
-private _exit = false;
-
-{
-    _x params ["", "_classID", "_partIndexN", "_amountOf", "_bleeding", "_damage", "_category"];
-    TRACE_2("OPENWOUND: ", _target, _x);
-
-    // Only parse injuries that are for the selected bodypart.
-    if (_partIndexN == _partIndex) then {
-        private _woundEffectiveness = _effectiveness;
-
-        // Select the classname from the wound classname storage
-        private _postfix = ["Minor", "Medium", "Large"] select _category;
-        private _className = format ["%1%2", EGVAR(medical_damage,woundClassNames) select _classID, _postfix];
-
-        // Check if this wound type has attributes specified for the used bandage
-        if (isClass (_config >> _className)) then {
-            // Collect the effectiveness from the used bandage for this wound type
-            private _woundTreatmentConfig = _config >> _className;
-
-            if (isNumber (_woundTreatmentConfig >> "effectiveness")) then {
-                _woundEffectiveness = getNumber (_woundTreatmentConfig >> "effectiveness");
-            };
-        } else {
-            //Basic medical bandage just has a base level config (same effectivenes for all wound types)
-            if (_bandage != "BasicBandage") then {
-                WARNING_2("No config for wound type [%1] config base [%2]", _className, _config);
-            };
-        };
-
-        TRACE_2("Wound classes: ", _specificClass, _classID);
-        if (_specificClass == _classID) exitWith {
-            _effectivenessFound = _woundEffectiveness;
-            _mostEffectiveSpot = _forEachIndex;
-            _mostEffectiveInjury = _x;
-            _exit = true;
-        };
-
-        // Check if this is the currently most effective found.
-        if (_woundEffectiveness * _amountOf * _bleeding > _effectivenessFound * (_mostEffectiveInjury select 3) * (_mostEffectiveInjury select 4)) then {
-            _effectivenessFound = _woundEffectiveness;
-            _mostEffectiveSpot = _forEachIndex;
-            _mostEffectiveInjury = _x;
-        };
-    };
-
-    if (_exit) exitWith {};
-} forEach _openWounds;
+private _targetWound = [_target, _bandage, _partIndex] call FUNC(findMostEffectiveWound);
```

Comment:
This change breaks being able to bandage a specific type of injury, due to the removal of:

```sqf
-        TRACE_2("Wound classes: ", _specificClass, _classID);
-        if (_specificClass == _classID) exitWith {
-            _effectivenessFound = _woundEffectiveness;
-            _mostEffectiveSpot = _forEachIndex;
-            _mostEffectiveInjury = _x;
-            _exit = true;
-        };
```

Reply:
So before removing this I looked back through the function calls and couldn't find anywhere that was using that parameter 🤔

Reply:
You are right - it's also not functional on the master branch. Seems like this is a bug we introduced somewhere along the way, though I couldn't find out at what point it broke, if it ever worked at all.. I do know it's intended to work through the medical menu. It was added as part of #172 in f6fbef980a.

Seeing as it's currently broken, I'm ok with removing this code and creating a seperate issue to investigate implementing the correct functionality for that (or decide on not doing it at all).

Reply:
Sounds good, my thoughts exactly